vault/token_store.go

```diff
@@ -87,6 +91,16 @@ func NewTokenStore(c *Core, config *logical.BackendConfig) (*TokenStore, error)
 	}
 	t.salt = salt
 
+	t.tokenLocks = map[string]*sync.RWMutex{}
+	for i := int64(0); i < 16; i++ {
```

Comment:
May as well do `for i:=0;i<256;i++ { key := fmt.Sprintf("%2x", i) }` to have only a single nesting layer